addons/l10n_it_edi_doi/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,94 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.fields import Command
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = 'sale.advance.payment.inv'
+
+    def _create_invoices(self, sale_orders):
+        """Extend to create a dedicated downpayment line for the declaration of intent amount."""
+        invoice = super()._create_invoices(sale_orders)
+
+        if self.advance_payment_method == 'delivered':
+            return invoice
+
+        order = self.sale_order_ids  # super calls ensure_one
+        doi_tax = order.l10n_it_edi_doi_id.company_id.l10n_it_edi_doi_tax_id
+        if not order.l10n_it_edi_doi_id or not doi_tax:
```

Comment:
No since all the fields on the "path" to `l10n_it_edi_doi_tax_id` (including `l10n_it_edi_doi_id`) are `Many2one` `doi_tax` will just be `False`.
I removed the `not order.l10n_it_edi_doi_id` since it is redundant (I even put it in a comment in the `if` before but then readded it ...)